mne/channels/channels.py

```diff
@@ -1007,11 +1011,134 @@ def _ch_neighbor_connectivity(ch_names, neighbors):
+    ch_names : list
+        The list of channel names present in connectivity matrix.
+
+    See Also
+    --------
+    read_ch_connectivity
+
+    Notes
+    -----
+    .. versionadded:: 0.15
+    """
+    (has_vv_mag, has_vv_grad, is_old_vv, has_4D_mag, ctf_other_types,
+     has_CTF_grad, n_kit_grads, has_any_meg, has_eeg_coils,
+     has_eeg_coils_and_meg, has_eeg_coils_only) = _get_ch_info(info)
+    conn_name = None
+    if has_vv_mag and ch_type == 'mag':
+        conn_name = 'neuromag306mag'
+    elif has_vv_grad and ch_type == 'grad':
+        conn_name = 'neuromag306planar'
+    elif ctf_other_types and ch_type == 'mag':
+        if info['nchan'] < 100:
+            conn_name = 'ctf64'
+        elif info['nchan'] > 200:
+            conn_name = 'ctf275'
+        else:
+            conn_name = 'ctf151'
+    elif has_4D_mag:
+        if 'MEG 248' in info['ch_names']:
+            idx = info['ch_names'].index('MEG 248')
+            grad = info['chs'][idx]['coil_type'] == FIFF.FIFFV_COIL_MAGNES_GRAD
+            mag = info['chs'][idx]['coil_type'] == FIFF.FIFFV_COIL_MAGNES_MAG
+            if ch_type == 'grad' and grad:
+                conn_name = 'bti248grad'
+            elif ch_type == 'mag' and mag:
+                conn_name = 'bti248'
+        elif 'MEG 148' in info['ch_names'] and ch_type == 'mag':
+            idx = info['ch_names'].index('MEG 148')
+            if info['chs'][idx]['coil_type'] == FIFF.FIFFV_COIL_MAGNES_MAG:
+                conn_name = 'bti148'
+
+    if conn_name is not None:
+        logger.info('Reading connectivity matrix for %s.' % conn_name)
+        return read_ch_connectivity(conn_name)
+    logger.info('Could not find a connectivity matrix for the data. '
+                'Computing connectivity based on Delaunay triangulations.')
+    return _compute_ch_connectivity(info, ch_type)
+
+
+def _compute_ch_connectivity(info, ch_type):
+    """Compute channel connectivity matrix using Delaunay triangulations.
+
+    Parameters
+    ----------
+    info : instance of mne.measuerment_info.Info
+        The measurement info.
+    ch_type : str
+        The channel type for computing the connectivity matrix. Currently
+        supports 'mag', 'grad' and 'eeg'.
+
+    Returns
+    -------
+    ch_connectivity : scipy.sparse matrix
+        The connectivity matrix.
+    ch_names : list
+        The list of channel names present in connectivity matrix.
+    """
+    from ..channels.layout import (_auto_topomap_coords, _find_neighbors,
+                                   _pair_grad_sensors)
+    combine_grads = (ch_type == 'grad' and FIFF.FIFFV_COIL_VV_PLANAR_T1 in
+                     np.unique([ch['coil_type'] for ch in info['chs']]))
+
+    if ch_type in ['mag', 'grad']:
+        picks = pick_types(info, meg=ch_type, ref_meg=False, exclude=[])
+    elif ch_type == 'eeg':
+        picks = pick_types(info, meg=False, eeg=True, ref_meg=False,
+                           exclude=[])
```

Comment:
Lines similar to the 8 above are in many places of mne - maybe there should be a helper function for that (sometime).